bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/callmonitor/CallMonitor.java

```diff
@@ -75,6 +75,9 @@ public CallMonitor(String ip, BoxHandler handler, ScheduledExecutorService sched
      */
     public void dispose() {
         reconnectJob.cancel(true);
+        if (monitorThread != null) {
+            monitorThread.interrupt();
+        }
```

Comment:
```suggestion
        CallMonitorThread monitorThread = this.monitorThread;
        if (monitorThread != null) {
            monitorThread.interrupt();
        }
```
Otherwise this results in a null-warning